src/controllers/customer.js

```diff
@@ -35,14 +35,48 @@ const createCustomer = async (req, res) => {
     // HTTP error codes: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses
     if (e instanceof PrismaClientKnownRequestError) {
       if (e.code === "P2002") {
-        return res.status(409).json({ error: "A customer with the provided email already exists" })
+        return res
+          .status(409)
+          .json({ error: "A customer with the provided email already exists" });
       }
     }
 
-    res.status(500).json({ error: e.message })
+    res.status(500).json({ error: e.message });
   }
-}
+};
+
+const updateCustomer = async (req, res) => {
+  const { id } = req.params;
+  const { name, contact } = req.body;
+
+  if (!name && !contact) {
+    return res.status(400).json({ error: "Missing field in request body" });
+  }
+
+  try {
+    const existingCustomer = await findCustomerDB(id);
+    if (!existingCustomer) {
+      return res
+        .status(404)
+        .json({ error: "Customer with that id does not exist" });
+    }
```

Comment:
If the database is set up correctly, it should already be throwing errors if you give invalid ids in your `UPDATE` and `INSERT` queries. You therefore don't need to check with a `SELECT` query before proceeding with trying to update.

Reply:
To add to that – you might have to look up the proper Prisma internal error key, I think it's P2025 – https://www.prisma.io/docs/orm/reference/error-reference#p2025

Reply:
hey, thank you both for the reference and feedback, I surely will update the code